### ISSUE TITLE
Userspace wifi frame crafting 

### ISSUE THREAD

**Issue by redfast00** [br101/libuwifi#5]
Hi, ESP8266 can send (almost) arbitrary packets. I am trying to implement EAP on an ESP8266, and I am looking for a library that makes it easy to craft 802.11 packets. I noticed there is no support for making packets (I think). Is making 802.11 packets in scope of this project?

**Comment by br101** [comment]
Yes, see https://github.com/br101/libuwifi/blob/master/core/inject.c

So far it only contains the two packet types I needed. Contributions welcome ;)

**Comment by redfast00** [comment]
Oh, hadn't noticed it. Will look into it when I have some spare time (around July/August). Thanks!

**Comment by warlock20** [comment]
Hallo @br101,

Now it has  `probe response` and `data frame`. Are you planning to write function to create IEEE802.11 frame for all types and subtypes. I.e separate function for each type?? (Current code structure looks like that)

**Comment by br101** [comment]
Well, i have just implemented what I needed at the time. A separate function for each frame type will result in a lot of functions, so if you have a better idea, please go for it... :)

**Comment by warlock20** [comment]
Exactly, current structure will result in lot of functions for each frame type. 
I have other ideas. Let me try it and will update after its done.

**Comment by redfast00** [comment]
@warlock20: I am curious to what you will implement. Any chance to get an EAP access point working?

**Comment by warlock20** [comment]
@redfast00 Haven't tried.. What you want to achieve?

**Comment by redfast00** [comment]
It would be really cool to be able to set up a rogue EAP access point to obtain hashes (like mana-hostapd but on an ESP8266).

**Comment by warlock20** [comment]
@redfast00  First I need to work on this and other features and I never used ESP8266 before. You can start the working and we will see.. :-)

**Comment by br101** [comment]
please note that frame injection does not work on esp8266 yet (but possible to do afaik)
